Only run coverage with inlining on

yuyichao opened the pull request:
Use a separate buildbot for noinline tests

Fix #55 
I realized that the test list is from `CoverageBase.jl` so this requires https://github.com/JuliaCI/CoverageBase.jl/pull/19 
@staticfloat

tkelman commented on master/nightly_noinline.py:
```diff
@@ -0,0 +1,73 @@
+noinline_nightly_scheduler = Nightly(name="Julia Noinline Build", builderNames=["nightly_noinline-x64"], hour=[3], branch="master", onlyIfChanged=True)
+c['schedulers'].append(noinline_nightly_scheduler)
+
+arch = "x86"
```
do we want this 32 or 64 bit?

yuyichao replied:
I make it run only on 64bit to save some time but deleted the wrong one here...... fixed....

tkelman commented:
can we make this run not on the ubuntu slave? there are too many nightly jobs on that builder and it gets backed up

yuyichao commented:
I've just been copying the previous config file I touched, where should I run this then?

tkelman commented:
centos 7 maybe? that only has the one job right now

yuyichao commented:
Sure, is this the right way to do it?

yuyichao commented:
I just realized that doing this will not run the tests with inliing off.

yuyichao commented:
@tkelman @staticfloat do we actually still want to have a buildbot running noinline tests? I think it won't really be testing sth that we want to keep working (now that we don't need inlining off for coverage). I might be useful as a Fuzz test but probably not more than that.

If we still want to have no-inline tests, https://github.com/JuliaLang/julia/pull/18680 should provide an much easier way.

tkelman commented:
I think it's worth identifying and tracking fragile tests, yes

yuyichao commented:
In that case https://github.com/JuliaLang/julia/pull/18681....

staticfloat commented:
It's trivial to add a second Ubuntu buildbot if that's what we need.  Then tasks assigned to an Ubuntu builder will be load-balanced accordingly.

tkelman commented:
we may want to do that anyway, but we have the centos worker and it's a bit underutilized

yuyichao commented:
Rebased and updated the script to use the env added in https://github.com/JuliaLang/julia/pull/18680. This still launches the test process manually so that the compile test on master process also has inlining off.
